### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by mjeffryes**
For some reason the index and install docs were not generated automatically. These are needed in order to publish to the registry, so I generated and added them separately for now. We'll still need to look into why they aren't getting generated so they don't get stale.

Workaround for: https://github.com/pulumi/pulumi-junipermist/issues/6

### CODE REVIEW

File: docs/installation-configuration.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      source = "registry.pulumi.io/juniper/mist"
+    }
+  }
+}
+
+provider "mist" {
+  host     = "api.mist.com"
+  apitoken = "xxxxxxxxxxxxxxxxxxxxxxxxxxx"
+}
+```
```
Comment (iwahbe) [inline docs/installation-configuration.md:53]: @mjeffryes This code isn't turned on for a reason. Of special significance, we are missing https://github.com/pulumi/pulumi-terraform-bridge/pull/2214.

Comment (mjeffryes) [reply docs/installation-configuration.md:53]: Ok, that makes perfect sense, but the real question is: How did these docs/* pages get here for other providers and why didn't that happen for this one? 

I checked other providers bridged recently (eg. dbtcloud) and these files we generated.

Comment (iwahbe) [reply docs/installation-configuration.md:53]: I have no idea. I thought that they were all hand authored. Until recently, the bridge didn't generate them.

Is it possible that you generated them the same way for dbtcloud as this provider, but the bridge just did something different. We are not maintaining stability guarantees on this feature yet.

Comment (mjeffryes) [reply docs/installation-configuration.md:53]: That's weird.

Eg. I made the initial commit for dbtcloud as well: [Initial commit · pulumi/pulumi-dbtcloud@9e22c8b](https://github.com/pulumi/pulumi-dbtcloud/commit/9e22c8be4d8a1cbc1c158aa278a77c19ae23e01e#diff-6ea303a35b3d9378eea0299496163e7c837b60501391c587a50dbc213038fce7) but definitely didn't hand author the files in docs/ for that provider.

**Pull request merged**